lib/index.js

```diff
@@ -1,75 +1,48 @@
 const deepmerge = require("deepmerge");
-const fs = require("fs");
-const path = require("path");
+const fs = require("node:fs");
+const path = require("node:path");
 const yaml = require("js-yaml");
 const prettier = require("prettier");
+const TYPES = require("./types");
+const UNITS = require("./cssUnits");
+const DEFAULT_CONFIG = require("./defaultConfig");
 const MESSAGES = require("./messages");
 
-const DEFAULT_CONFIG = {
-  extractTo: {
-    css: true,
-    js: true,
-  },
-};
-
 module.exports = class Breakpoints {
   /**
-   * @type {object}
-   * @property {object} paths - Relativ to the themes root folder ...
-   * @property {string} paths.breakpoints - ...path to drupals breakpoint file
-   * @property {string} paths.css - ...path to your final css file
-   * @property {string} paths.js - ...path to your final js file
-   * @property {string} themeName - Drupals custom theme name
-   * @property {string} extractTo -
-   * @property {boolean} [js] - (default true), should generate js file?
-   * @property {boolean} [css] - (default true), should generate css file?
+   * @type {TYPES.USER_CONFIG}
    */
-  #userConfig;
+  #userConfig = DEFAULT_CONFIG;
 
   /**
-   * @typedef {object} BREAKPOINT
-   * @property {string} label -
-   * @property {string} mediaQuery -
-   * @property {number} weight -
-   * @property {string[]} multipliers -
-   * @property {string} [group] -
+   * @type {Object<string, TYPES.BREAKPOINT>}
    */
-
-  /**
-   * @type {Object<string, BREAKPOINT>}
-   */
-  #breakpoints;
+  #breakpoints = {};
 
   /**
    * @type {string[]}
    */
-  #groups;
+  #groups = [];
 
   /**
-   * @type {Map<string, BREAKPOINT[]>}
+   * @type {Map<string, TYPES.BREAKPOINT[]>}
    */
-  #breakpointsMap;
+  #breakpointsMap = new Map();
 
   /**
    * @type {Array}
    */
-  #customProperties;
+  #customProperties = [];
 
   /**
    * @type {string}
    */
-  #userConfigName;
```

Comment:
I would use `breakpoints.config.yml` (without the dot in front). It looks like not using dotfiles for config anymore is the path a lot of tools move to, e.g.:

* https://vitejs.dev/config/
* https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new
* https://nuxtjs.org/docs/directory-structure/nuxt-config